src/RestController.php

```diff
@@ -998,4 +973,46 @@ protected function createHalEntity($entity)
             $this->getRouteIdentifierName()
         );
     }
+
+    /**
+     * @param Exception $exception
+     * @return ApiProblem
+     */
+    private function createApiProblemFromException(Exception $exception)
+    {
+        return new ApiProblem(
+            $this->validateHttpStatusCode($exception->getCode()),
+            $exception->getMessage()
+        );
+    }
+
+    /**
+     * @param Throwable $throwable
+     * @return ApiProblem
+     */
+    private function createApiProblemFromThrowable(Throwable $throwable)
```

Comment:
There's no reason to split this functionality into two separate methods, particularly as, with private visibility, the methods are clearly internal. Combine them into one, and remove the typehint.